common/http.h

```diff
@@ -0,0 +1,73 @@
+#pragma once
+
+#include <cpp-httplib/httplib.h>
+
+struct common_http_url {
+    std::string scheme;
+    std::string user;
+    std::string password;
+    std::string host;
+    std::string path;
+};
+
+static common_http_url common_http_parse_url(const std::string & url) {
```

Comment:
Seems like the current parser does not take into account query params. I think it can be better if we can make it [rfc3986](https://datatracker.ietf.org/doc/html/rfc3986)-compliant, some test cases are here: https://github.com/jholloc/simple-uri-parser/blob/main/tests.cpp